src/unitxt/splitters.py

```diff
@@ -108,9 +108,7 @@ class RandomSampler(Sampler):
 
     def prepare(self):
         super().prepare()
-        self.random_generator = get_sub_default_random_generator(
-            sub_seed="random_sample_seed"
-        )
+        self.random_generator = new_random_generator(sub_seed="random_sample_seed")
```

Comment:
This is in the general score of the operator which means:
1) Calling the same operator twice in a row will affect each other and won't be identical
2) Calling different streams from the operators in different order will affect its behavior per that stream

Reply:
oh I thought there was some level of independence here with the generator is reinitialized in `prepare()`.
when is `prepare` called?

the alternative is having the generator initialized in sample(), right? then it can depend on the `instances_pool`, but isn't that pool fixed?

Reply:
The prepare is called in the instantiation which means once in a class life time. 

The rest lets discuss

Reply:
fixed